File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> res  =  new ArrayList<>();
        Queue<TreeNode> q =  new LinkedList<>();

        if(root == null) {
            return new ArrayList<>();
        }
        int Level =  0;

        q.offer(root);
        while(!q.isEmpty()) {
            List<Integer> ans =  new ArrayList<>();
            Level++;
            int size = q.size();
            for(int i = 0; i < size ; i++) {
                TreeNode temp = q.poll();
                ans.add(temp.val);

                if(temp.left != null) {
                    q.offer(temp.left);
                }
                if(temp.right!= null) {
                    q.offer(temp.right);
                }
            }
            int n = ans.size() - 1;  
            res.add(ans.get(ans.size()-1));
        }
        return res;
    }
}

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left),
 * right(right) {}
 * };
 */
class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        vector<int> res;
        queue<TreeNode*> q;
        int level = 0;

        if (root == NULL)
            return res;
        q.push(root);

        while (!q.empty()) {
            vector<int> vis;   // store values of each level //
            int size = q.size();
            level++;
            for (int i = 0; i < size; i++) {
                TreeNode* node = q.front();
                q.pop();
                vis.push_back(node->val);

                if (node->left)
                    q.push(node->left);
                if (node->right)
                    q.push(node->right);
            }
            int temp = vis[vis.size() - 1];
            res.push_back(temp);
        }
        return res;
    }
};